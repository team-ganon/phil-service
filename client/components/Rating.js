import React from 'react';
import styles from '../CSS/rating.css';

const Rating = ({name, score}) => {
  let title = name.name;
  let style = {
    width: `${score * 2}0%`
  };
  return (
    <div className={styles.ratingDiv}>
      <p className={styles.title}>{title}</p>
      <div className={styles.ratingBarAndScore}>
        <div className={styles.outerLineDiv}>
          <div className={styles.innerLineDiv} style={style}></div>
        </div>
        <p className={styles.score}>{score}</p>
      </div>
    </div>
  );
};

export default Rating;