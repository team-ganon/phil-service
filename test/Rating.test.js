import React from 'react';
import renderer from 'react-test-renderer';
import Rating from '../client/components/Rating';


describe('Rating Component', () => {

  it('Rating SnapShot Test', () => {
    const RatingJson = renderer.create(<Rating name={{name: 'Cleanliness'}} score={3} />).toJSON();
    expect(RatingJson).toMatchSnapshot();
  });
  
});

