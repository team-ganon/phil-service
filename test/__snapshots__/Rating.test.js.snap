// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Component Rating SnapShot Test 1`] = `
<div
  className="ratingDiv"
>
  <p
    className="title"
  >
    Cleanliness
  </p>
  <div
    className="ratingBarAndScore"
  >
    <div
      className="outerLineDiv"
    >
      <div
        className="innerLineDiv"
        style={
          Object {
            "width": "60%",
          }
        }
      />
    </div>
    <p
      className="score"
    >
      3
    </p>
  </div>
</div>
`;
